src/main/webapp/app/code-editor/ace/code-editor-ace.component.ts

```diff
@@ -95,13 +91,18 @@ export class CodeEditorAceComponent implements OnInit, AfterViewInit, OnChanges,
      * @param {SimpleChanges} changes
      */
     ngOnChanges(changes: SimpleChanges): void {
-        if (changes.participation && changes.participation.currentValue) {
-            this.updateFileChannel = `/topic/repository/${this.participation.id}/file`;
-            this.receiveFileUpdatesChannel = `/user${this.updateFileChannel}`;
+        if (hasParticipationChanged(changes)) {
+            this.updateUnsavedFilesChannel = `/topic/repository/${this.participation.id}/files`;
+            this.receiveFileUpdatesChannel = `/user${this.updateUnsavedFilesChannel}`;
             this.setUpReceiveFileUpdates();
         }
-        if (changes.participation && this.participation) {
-            this.updateSaveStatusLabel();
+        // Update editor file session object to include new files and remove old files
+        if (changes.repositoryFiles) {
+            const newFiles: string[] = _difference(changes.repositoryFiles.currentValue, changes.repositoryFiles.previousValue);
```

Comment:
why does `_difference` starts with an underscore?

Reply:
This is a convention I follow because lodash/rxjs share a lot of functions with the same name:
- lodash normal functions with underscore
- rxjs with rx prefix if needed
- rxjs fp without any prefix